### PR TITLE
fix(tls): tls configuration now completes after TLS configuration

### PR DESCRIPTION
**Pull request by rsdmike**

### CODE REVIEW

File: internal/lm/engine.go [hunk 1]
```diff
@@ -58,20 +58,28 @@ func (lme *LMEConnection) Initialize() error {
...
-	lme.ourChannel = ((lme.ourChannel + 1) % 32) + 1
-	// err := lms.execute()
+	lme.ourChannel = ((lme.ourChannel + 1) % 32)
+
 	bin_buf := apf.ChannelOpen(lme.ourChannel)
 	err := lme.Command.Send(bin_buf.Bytes(), uint32(bin_buf.Len()))
 	if err != nil {
-		log.Fatal(err)
+		if err.Error() == "no such device" || err.Error() == "The device is not connected." {
+			// i mean why not
```
Comment (rjbrache) [inline internal/lm/engine.go:67]: Unclear comment

File: internal/lm/engine.go [hunk 2]
```diff
@@ -58,20 +58,28 @@ func (lme *LMEConnection) Initialize() error {
...
+			err = lme.Initialize()
+			if err == nil {
+				return lme.Connect()
+			}
+		} else {
```
Comment (rjbrache) [inline internal/lm/engine.go:72]: Do we not want to log the error if Initialize fails?

File: pkg/apf/processor.go [hunk 3]
```diff
@@ -42,7 +42,8 @@ func Process(data []byte, session *LMESession) bytes.Buffer {
 		break
 	case APF_CHANNEL_CLOSE: // (97) Intel AMT is closing this channel, we need to disconnect the LMS TCP connection
 		log.Debug("received APF_CHANNEL_CLOSE")
-		dataToSend = ProcessChannelClose(data, session)
+		// dataToSend =
+		ProcessChannelClose(data, session)
```
Comment (rjbrache) [inline pkg/apf/processor.go:46]: Why are we keeping this line in the code if it is commented out?

File: internal/client/executor.go [hunk 4]
```diff
@@ -78,14 +78,15 @@ func (e Executor) MakeItSo(messageRequest rps.Message) {
 	if e.status != nil {
 		defer close(e.status)
 	}
+
+	//go e.localManagement.Listen()
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply internal/client/executor.go]: removed

File: internal/lm/engine.go (hunk 1 above)
Comment (rsdmike) [reply internal/lm/engine.go]: updated

File: pkg/apf/processor.go (hunk 3 above)
Comment (rsdmike) [reply pkg/apf/processor.go:45]: removed

File: internal/lm/engine.go [hunk 5]
```diff
@@ -58,20 +58,28 @@ func (lme *LMEConnection) Initialize() error {
...
 	bin_buf := apf.ChannelOpen(lme.ourChannel)
 	err := lme.Command.Send(bin_buf.Bytes(), uint32(bin_buf.Len()))
 	if err != nil {
-		log.Fatal(err)
+		if err.Error() == "no such device" || err.Error() == "The device is not connected." {
+			// i mean why not
+			err = lme.Initialize()
+			if err == nil {
+				return lme.Connect()
+			}
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply internal/lm/engine.go:71]: I can add a retry comment

**Review by rjbrache — APPROVED** [review]

**Pull request merged**